replica/out_of_space_controller.cc

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2025-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+ */
+#include "replica/out_of_space_controller.hh"
+#include "utils/log.hh"
+
+namespace replica {
+
+static logging::logger logger("out_of_space_controller");
+
+future<void> set_database_in_critical_disk_utilization_mode(sharded<database>& db, bool enabled)
+{
+    logger.info("{} database critical disk utilization mode (rejecting user writes)", enabled ? "enabling" : "disabling");
```

Comment:
We now do more than rejecting user writes.
Do we want to list all actions taken, none, or some?